configure.ac

```diff
@@ -69,7 +69,7 @@ AC_SUBST([ldsoconfdir], [/etc/ld.so.conf.d])
 AC_SUBST([pkgconfigdir], [${libdir}/pkgconfig])
 
 AC_PATH_KERNEL_SOURCE
-AC_KERNEL_CHECKS
+AS_IF([test $build_kernel_module = 1], [AC_KERNEL_CHECKS])
```

Comment:
Since `AC_PATH_KERNEL_SOURCE` already checks for `$build_kernel_module`, maybe rename and move `AC_KERNEL_CHECKS` under `AC_PATH_KERNEL_SOURCE`, also while at it, fix corresponding help message that does not work?

```diff
diff --git a/configure.ac b/configure.ac
index e17c7bc..d266030 100644
--- a/configure.ac
+++ b/configure.ac
@@ -68,7 +68,6 @@ AC_SUBST([initdir], [${sysconfdir}/init.d])
 AC_SUBST([ldsoconfdir], [/etc/ld.so.conf.d])
 AC_SUBST([pkgconfigdir], [${libdir}/pkgconfig])

-AC_PATH_KERNEL_SOURCE
 AC_KERNEL_CHECKS

 AC_CONFIG_FILES([Makefile
diff --git a/m4/ac_path_kernel_source.m4 b/m4/ac_path_kernel_source.m4
index 40d12e9..5877533 100644
--- a/m4/ac_path_kernel_source.m4
+++ b/m4/ac_path_kernel_source.m4
@@ -42,13 +42,17 @@ AC_DEFUN([AC_PATH_KERNEL_SOURCE_SEARCH],
 ]
 )

-AC_DEFUN([AC_PATH_KERNEL_SOURCE],
+AC_DEFUN([AC_KERNEL_CHECKS],
 [
   AC_CHECK_PROG(ac_pkss_mktemp,mktemp,yes,no)
-  AC_PROVIDE([AC_PATH_KERNEL_SOURCE])
+  AC_PROVIDE([AC_KERNEL_CHECKS])
+
+  AC_ARG_ENABLE([kernel-module],
+    [AS_HELP_STRING([--disable-kernel-module],
+                    [Disable building the kernel module (Default is enabled)])],
+    [build_kernel_module=$enableval],
+    [build_kernel_module=1])

-  AC_ARG_ENABLE([kernel-module],[Enable building the kernel module (default: enabled)],[build_kernel_module=$enableval],
-               [build_kernel_module=1])
   AS_IF([test $build_kernel_module = 1],[

   AC_MSG_CHECKING([for Linux kernel sources])
@@ -88,12 +92,14 @@ AC_DEFUN([AC_PATH_KERNEL_SOURCE],

   AC_MSG_CHECKING([kernel release])
   AC_MSG_RESULT([${kernelvers}])
+
+  AC_KERNEL_CHECK_SUPPORT
   ])
   AM_CONDITIONAL([BUILD_KERNEL_MODULE], [test $build_kernel_module = 1])
 ]
 )

-AC_DEFUN([AC_KERNEL_CHECKS],
+AC_DEFUN([AC_KERNEL_CHECK_SUPPORT],
 [
   srcarch=$(uname -m | sed -e s/i.86/x86/ \
                            -e s/x86_64/x86/ \
```

Reply:
This seems like a possibly better approach. I will implement this.

Reply:
Done and tested